web/impact/impact/tests/test_create_edit_office_hours_view.py

```diff
@@ -0,0 +1,244 @@
+
+    def test_none_staff_or_none_mentor_response(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        response = self._create_office_hour_session(self.basic_user(), data)
+        self.assertEqual(response.data['detail'],
+                         DEFAULT_PERMISSION_DENIED_DETAIL)
+
+    def test_use_request_user_for_mentor_users(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        mentor2 = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor,
+                                           get_data={'mentor': mentor2.id})
+        response = self._create_office_hour_session(mentor, data)
+        self.assertEqual(response.data['data']['mentor']['id'], mentor.id)
+
+    def test_admin_cant_create_office_hour_for_restricted_user_response(self):
+        user = self._expert_user(UserRole.JUDGE)
+        data = self._get_post_request_data(user)
+        response = self._create_office_hour_session(self.staff_user(), data)
+        self._assert_error_response(response.data,
+                                    key='mentor', expected=INVALID_USER)
+
+    def test_admin_cant_create_office_hour_for_restricted_user(self):
+        user = self._expert_user(UserRole.JUDGE)
+        data = self._get_post_request_data(user)
+        with self._assert_office_hour_created(created=False):
+            self._create_office_hour_session(self.staff_user(), data)
+
+    def _expert_with_inactive_program(self, role):
+        program = ProgramFactory(program_status='ended')
+        return self._expert_user(role, program)
+
+    def _get_post_request_data(self,
+                               mentor, minutes_from_now=0, get_data=None):
+        location = LocationFactory()
+        start_time = datetime.now() + timedelta(minutes=minutes_from_now)
+        data = {
+            'mentor': mentor.id,
+            'start_date_time': start_time,
+            'end_date_time': start_time + timedelta(minutes=30),
+            'topics': 'topics',
+            'description': 'description',
+            'location': location.id
+        }
+        if get_data:
+            data.update(get_data)
+        return data
+
+    def _assert_success_response(self, data, edit=False):
+        header = SUCCESS_EDIT_HEADER if edit else SUCCESS_CREATE_HEADER
+        self.assertTrue(all([
+            data['success'],
+            data['header'] == header,
+        ]))
+
+    def _assert_fail_response(self, data, edit=False):
+        header = FAIL_EDIT_HEADER if edit else FAIL_CREATE_HEADER
+        self.assertTrue(all([
+            not data['success'],
+            data['header'] == header,
+        ]))
+
+    def _assert_error_response(self, data, key, expected):
+        self.assertIn(expected, data['errors'][key])
+
+    def _create_office_hour_obj(self,
+                                mentor, minutes_from_now=0, finalist=None):
```

Comment:
let's put these args on separate lines (since we've had to break once, break on each)